forem/forem
# Auto generate mascot accounts

## What type of PR is this? (check all applicable)

## Description
This PR removes the `SiteConfig` requirement of `mascot_user_id` by auto-generating a mascot account when setting up a Forem.

Eventually, we'll want to 1) move some more mascot fields to `SiteConfig` and 2) refactor Mascot altogether into its own domain.

## Related Tickets & Documents
closes https://github.com/forem/InternalProjectPlanning/issues/94

## QA Instructions, Screenshots, Recordings
1. Set `ENV["MODE"]` in your environment to `"STARTER"`.
2. Reset your DB (`rails db:reset`) - **warning** this will erase your local database.
3. Fire up the app locally.
4. Go through onboarding, create an account.
5. Fire up a console and verify that `User.mascot_account` returns a mascot user.

## Added tests?

## Added to documentation?


![blank_face_with_smoke_gif](https://media.giphy.com/media/24pBw18bgHMPu/giphy.gif)
